app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java

```diff
@@ -89,60 +90,53 @@ private boolean operandEnabled(String operand) {
     }
 
     private boolean operationEnabled() {
-        return (operandEnabled(firstOperand) && operandEnabled(secondOperand) && !operator.isEmpty());
+        boolean operationEnabled = true;
+        if (getError() == Error.ERROR_DIVISION_BY_ZERO || getError() == Error.ERROR_INCOMPLETE_OPERATION) {
+            operationEnabled = false;
+        }
+        return operationEnabled;
+    }
+
+
+    private Double makeOperation() {
+        switch (operator) {
+            case ADD: {
+                return Double.parseDouble(firstOperand) + Double.parseDouble(secondOperand);
+            }
+            case SUB: {
+                return Double.parseDouble(firstOperand) - Double.parseDouble(secondOperand);
+
+            }
+            case MUL: {
+                return Double.parseDouble(firstOperand) * Double.parseDouble(secondOperand);
+            }
+            case DIV: {
+                return Double.parseDouble(firstOperand) / Double.parseDouble(secondOperand);
+            }
+        }
+        return null;
```

Comment:
here you have the same problem returned a null...please use 0.0 or something

Reply:
Solved!